tests/test_routes.py

```diff
@@ -0,0 +1,158 @@
+from schematic.manifest.generator import ManifestGenerator
+
+from unittest.mock import patch
+
+
+@pytest.fixture(scope="class")
+def app():
+    app = create_app()
+    yield app
+
+
+@pytest.fixture(scope="class")
+def data_model_jsonld():
+    data_model_jsonld = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+    yield data_model_jsonld
+
+
+def test_manifest_generation_load_config(app, config_path, data_model_jsonld):
+    """Test if asset_view gets updated"""
+    app.config["SCHEMATIC_CONFIG"] = config_path
+    mg = ManifestGeneration(
+        schema_url=data_model_jsonld,
+        data_type=["test_data_type"],
+        asset_view="syn123",
+    )
+    config = mg._load_config_(app=app)
+
+    assert config["synapse"]["master_fileview"] == "syn123"
+
+
+def test_check_dataset_match_data_type(data_model_jsonld):
+    """Test if function could raise an error when number of data types do not match number of dataset ids"""
+    with pytest.raises(ValueError):
+        mg = ManifestGeneration(
+            schema_url=data_model_jsonld,
+            data_type=["Biospecimen", "Patient"],
+            asset_view="syn123",
+            dataset_id=["syn1234"],
+        )
+
+
+def test_check_if_asset_view_valid(data_model_jsonld):
+    with pytest.raises(ValueError):
+        mg = ManifestGeneration(
+            schema_url=data_model_jsonld,
+            data_type=["Biospecimen", "Patient"],
+            asset_view="invalid asset view id",
+            dataset_id=["syn1234"],
+        )
+
+
+def test_check_if_dataset_id_valid(data_model_jsonld):
+    with pytest.raises(ValueError):
+        mg = ManifestGeneration(
+            schema_url=data_model_jsonld,
+            data_type=["Biospecimen", "Patient"],
+            asset_view="syn1234",
+            dataset_id=["invalid dataset id"],
+        )
+
+
+@pytest.mark.parametrize(
+    "test_title, mock_datatype, expected_title",
+    [
+        ("Mock", "Biospecimen", "Mock.Biospecimen.manifest"),
+        (None, "Patient", "Example.Patient.manifest"),
+    ],
+)
+def test_get_manifest_title(
+    test_title, mock_datatype, expected_title, data_model_jsonld
+):
+    """Test if title of manifest gets updated correctly"""
+    mg = ManifestGeneration(
+        schema_url=data_model_jsonld, data_type=[mock_datatype], title=test_title
+    )
+    title = mg._get_manifest_title(single_data_type=mock_datatype)
+    assert title == expected_title
+
+
+@pytest.mark.parametrize("output_format", ["google_sheet", "dataframe"])
+def test_create_single_manifest(output_format, data_model_jsonld):
+    """create_single_manifest wraps around manifest_generator.get_manifest function.
+    TODO: add test for output_format="excel"
+    """
+    mg = ManifestGeneration(
+        schema_url=data_model_jsonld, data_type=["Patient"], output_format=output_format
+    )
+
+    if output_format == "google_sheet":
```

Comment:
If you need to include conditionals like this in a parameterized test, that suggests that the interface that you're testing should be refactored, or the test should be split. Maybe you can make a fixture for the `ManifestGeneration()` instance to avoid duplicating that bit of code.

Reply:
Thanks for the advice Bruno! I created a fixture for the ManifestGeneration() and separated the test